### ISSUE TITLE
Drawing galaxies on tiles is really slow

### ISSUE THREAD

**Issue by ismael-mendoza** [prob-ml/bliss#110]
After trying to fix my PR #99  this morning I realized that the current master branch has a problem where it is really slow to draw galaxies on tiles. This problem doesn't happen with stars (set `prob_galaxy = 0.0` ) so this seems like a bug. This is even after the fix in #109 where we can now form images with tiles that have no padding. 

@Runjing-Liu120 what do you think it could be ? 

Here is a minimal example reproducing the problem: 
```
from pathlib import Path
import torch
import numpy as np
from bliss.datasets import simulated

use_cuda = torch.cuda.is_available()
device = torch.device("cuda" if use_cuda else "cpu")
data_path = root_path.joinpath("data")

slen = 50
n_bands = 1

# setup dataset
background = torch.zeros(1, slen, slen, device=device)
background[0] = 686.0
psf_file = data_path.joinpath("fitted_powerlaw_psf_params.npy")
psf_params = torch.from_numpy(np.load(psf_file)).to(device)[range(n_bands)]

dec_file = data_path.joinpath("galaxy_decoder_1_band.dat")
dec = simulated.SimulatedDataset.get_gal_decoder_from_file(dec_file)
dec_args = (dec, psf_params, background)
dec_kwargs = {
    "prob_galaxy": 1.0, 
    "n_bands": n_bands,
    "slen": slen,
    "tile_slen": 2,
    "max_sources_per_tile": 3,
    "mean_sources_per_tile": 1.5,
    "min_sources_per_tile": 0,
    "ptile_padding": 0,
}

n_batches = 2
batch_size = 2
dataset = simulated.SimulatedDataset(n_batches, batch_size, dec_args, dec_kwargs)

# generate batches of data
# this is *really* slow for prob_galaxy = 1.0 and tile_slen = 2 above
for i in range(10):
    dataset.get_batch()
```

You can try switching `prob_galaxy= 0.0` and `tile_slen = slen` above to see the issue.
